sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -136,6 +136,7 @@ class BlobClient(StorageAccountHostsMixin, StorageEncryptionMixin):  # pylint: d
         or 4MB.
     :keyword str version_id: The version id parameter is an opaque DateTime value that, when present,
         specifies the version of the blob to operate on.
+    :keyword BlobTokenAudience audience: The Azure Active Directory Audience to use when authorizing requests.
```

Comment:
Maybe add something about when using AAD auth via a TokenCredential. Maybe even a specific note about only has an effect when credential is of type TokenCrednetial.

"The audience to use when requesting tokens for Azure Active Directory authentication. Only has an effect when credential is of type TokenCredential"